### PR TITLE
STOR-1600: Rebase against v3.2.0

### PR DESCRIPTION
**Pull request by gnufied**
This PR rebases against v3.2.0.

Following commits had to be manually cherry-pick after `git merge -s ours` strategy:

```
        c1f934582                       UPSTREAM: <carry>: Add OpenShift files  https://github.com/openshift/vmware-vsphere-csi-driver/commit/c1f934582?w=1
        7e503e5af                       UPSTREAM: <carry>: Updating vmware-vsphere-syncer-container image to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/9dc2a375bf335e7e4e3d7e75abc5c3bdbb0c0e98/images/vmware-vsphere-syncer.yml https://github.com/openshift/vmware-vsphere-csi-driver/commit/7e503e5af?w=1
        7cfd0cbfa                       UPSTREAM: <carry>: Updating ose-vmware-vsphere-csi-driver-container image to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/9dc2a375bf335e7e4e3d7e75abc5c3bdbb0c0e98/images/ose-vmware-vsphere-csi-driver.yml https://github.com/openshift/vmware-vsphere-csi-driver/commit/7cfd0cbfa?w=1
        2c7016be9                       UPSTREAM: <carry>: Updating vmware-vsphere-syncer-container image to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/e5df10e41b14c724468c8f78e2690f219279bfc4/images/vmware-vsphere-syncer.yml https://github.com/openshift/vmware-vsphere-csi-driver/commit/2c7016be9?w=1
        acdaf7289                       UPSTREAM: <carry>: STOR-1688: Chore: add .snyk file to ignore false positives   https://github.com/openshift/vmware-vsphere-csi-driver/commit/acdaf7289?w=1
        fb633489a                       UPSTREAM: <carry>: Updating vmware-vsphere-syncer-container image to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/4022cd290f00a44d667dda03f2d78d84a488c7ed/images/vmware-vsphere-syncer.yml https://github.com/openshift/vmware-vsphere-csi-driver/commit/fb633489a?w=1
        4aa7d9764                       UPSTREAM: <carry>: Updating ose-vmware-vsphere-csi-driver-container image to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/4022cd290f00a44d667dda03f2d78d84a488c7ed/images/ose-vmware-vsphere-csi-driver.yml https://github.com/openshift/vmware-vsphere-csi-driver/commit/4aa7d9764?w=1
        7d774f355                       UPSTREAM: <carry>: Add vendor   https://github.com/openshift/vmware-vsphere-csi-driver/commit/7d774f355?w=1
        b2d9fc726                       UPSTREAM: 2686: Bump OpenTelemetry libs (#2686) https://github.com/openshift/vmware-vsphere-csi-driver/commit/b2d9fc726?w=1     https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2686
        06e5dd899                       UPSTREAM: <drop>: Add vendored changes  https://github.com/openshift/vmware-vsphere-csi-driver/commit/06e5dd899?w=1
        424b6b398                       UPSTREAM: <carry>: OCPBUGS-27242: fix or ignore snyk errors for ocp storage repos       https://github.com/openshift/vmware-vsphere-csi-driver/commit/424b6b398?w=1
        0e8070baa                       UPSTREAM: 2820: CVE-2024-24786: bump google.golang.org/protobuf to v1.33.0      https://github.com/openshift/vmware-vsphere-csi-driver/commit/0e8070baa?w=1     https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2820
        a11407851                       UPSTREAM: 2814: Fix volume provisioning when duplicate region or zone tags exist        https://github.com/openshift/vmware-vsphere-csi-driver/commit/a11407851?w=1
  https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2814
        96934fcba                       UPSTREAM: 2593: GetDevMounts: eval symlinks found in mount table        https://github.com/openshift/vmware-vsphere-csi-driver/commit/96934fcba?w=1     https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2593
```

### CODE REVIEW

**Comment by dobsonj** [comment]
@gnufied do you really need to keep this cherrypick?
```
        b2d9fc726                       UPSTREAM: 2686: Bump OpenTelemetry libs (#2686) https://github.com/openshift/vmware-vsphere-csi-driver/commit/b2d9fc726?w=1     https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2686
```
The otel library bump was included in v3.2.0 and the commit just has an empty line in go.mod:
https://github.com/openshift/vmware-vsphere-csi-driver/pull/122/commits/a568e8b4a87f90e0298724035dd788ad2fe0856c

That's a really minor thing though, I'll leave it up to you if you want to drop that commit or not. The rest looks good to me.
/lgtm

For reference I mostly just compared the patches in this rebase with what we had in 4.16.
4.16: https://github.com/kubernetes-sigs/vsphere-csi-driver/compare/v3.1.2...openshift:release-4.16
4.17: https://github.com/kubernetes-sigs/vsphere-csi-driver/compare/v3.2.0...gnufied:upstream-3.2

**Comment by mpatlasov** [comment]
/lgtm

I've got stuck on the same (minor) issue as Jonathan mentioned above -- it's not clear why we need to explicitly cherry-pick "UPSTREAM: 2686: Bump OpenTelemetry libs". Another minor concern: If we're going to cherry-pick "Add OpenShift files" again and again in future, the list of ART changes will grow more and more as well. I think it's simpler to create new ""Add OpenShift files" commit from scratch copy/paste-ing our OCP files (including `.snyk`) and removing `.github`.

**Comment by gnufied** [comment]
I squashed all ART commits into one. Yeah so `opentelemetry` changes all really boil down to vendored changes - https://github.com/openshift/vmware-vsphere-csi-driver/pull/122/commits/a568e8b4a87f90e0298724035dd788ad2fe0856c

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/evanphx/json-patch/v5 v5.6.0
 	github.com/fsnotify/fsnotify v1.7.0
-	github.com/golang/protobuf v1.5.3
+	github.com/golang/protobuf v1.5.4
```
Comment (RomanBednar) [inline go.mod:12]: Why would this be a separate `<drop>` commit? We have a [PR](https://github.com/kubernetes-sigs/vsphere-csi-driver/commit/862c6568b7d14254f5a4820dc2a91c8554fc1b7e) upstream that bumps the versions so I'd expect this to be included in one commit: `UPSTREAM: 2820: bump google.golang.org/protobuf to v1.33.0`

And vendoring can be done in a single commit to keep this cleaner, currently the rebase updates vendor in 3 different commits - do we actually need that?

File: .snyk [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# References:
```
Comment (RomanBednar) [inline .snyk]: This commit could be also squashed into `UPSTREAM: <carry>: Add OpenShift files`

File: .ci-operator.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.20-openshift-4.15
+  tag: rhel-9-release-golang-1.21-openshift-4.16
```
Comment (RomanBednar) [inline .ci-operator.yaml]: This commit could be also squashed into `UPSTREAM: <carry>: Add OpenShift files`

**Comment by dobsonj** [comment]
Thanks @gnufied 
/lgtm

**Pull request merged**